src/connector.rs

```diff
@@ -24,7 +25,7 @@ pub struct HttpsConnector<T> {
     force_https: bool,
     http: T,
     tls_config: Arc<rustls::ClientConfig>,
-    override_server_name: Option<String>,
+    server_name_resolver: Option<Arc<dyn ResolveServerName + Sync + Send>>,
```

Comment:
Not sure if it still makes sense to wrap this in `Option`? I think we should just have a `FromUriResolver` that is used by default?

I think that could also address the duplication of the IPv6 stripping.

Reply:
Sure, will add a default resolver.